router/ingress.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2016 The Fission Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package router
+
+import (
+	"log"
+	"os"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+
+	"github.com/fission/fission/crd"
+)
+
+var podNamespace string
+
+func init() {
+	podNamespace = os.Getenv("POD_NAMESPACE")
+	if podNamespace == "" {
+		podNamespace = "fission"
+	}
+}
+
+func createIngress(trigger *crd.HTTPTrigger, kubeClient *kubernetes.Clientset) {
+
+	if !trigger.Spec.CreateIngress {
+		log.Printf("Skipping creation of ingress for trigger: %v", trigger.Metadata.Name)
+	}
+
+	ing := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:       getDeployLabels(trigger),
+			GenerateName: "ingress-" + trigger.Spec.FunctionReference.Name + "-",
+			// The Ingress NS MUST be same as Router NS, check long discussion:
+			// https://github.com/kubernetes/kubernetes/issues/17088
+			// We need to revisit this in future, once Kubernetes supports cross namespace ingress
+			Namespace: podNamespace,
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: trigger.Spec.Host,
```

Comment:
pardon my ignorance here but i dont understand the actual use of "host" parameter in the context of a normal http trigger. but now with the ingress using it, does it mean the router can be reachable using the host FQDN?

Reply:
I don't fully understand it, but if you map your LB to a DNS name, then the `host` becomes useful.